pkg/kube/controllers/extendedstatefulset/reconciler.go

```diff
@@ -258,3 +282,77 @@ func (r *ReconcileExtendedStatefulSet) getActualStatefulSet(ctx context.Context,
 
 	return result, maxVersion, nil
 }
+
+// listStatefulSetVersions gets all StatefulSets' versions and ready status owned by the ExtendedStatefulSet
+func (r *ReconcileExtendedStatefulSet) listStatefulSetVersions(ctx context.Context, exStatefulSet *essv1a1.ExtendedStatefulSet) (map[int]bool, error) {
```

Comment:
I know that `listStatefulSets` was already defined on `*ReconcileExtendedStatefulSet`, but I wonder if it would make sense to move these functions to the `ExtendedStatefulSet` type instead. They are not really `Reconciler` domain imo.